doc/cygnus-ngsi-ld/flume_extensions_catalogue/ngsi_rest_handler.md

```diff
@@ -0,0 +1,142 @@
+
+```
+notification={
+   headers={
+	   fiware-service=hotel1,
+	   link=https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld
+	   correlation-id=1234567890-0000-1234567890
+   },
+   body={
+      {
+	      entityId=suite.12,
+	      entityType=room,
+	      attributes=[
+	         ...
+	      ]
+	   },
+	   {
+	      entityId=other.9,
+	      entityType=room,
+	      attributes=[
+	         ...
+	      ]
+	   }
+	}
+}
+```
+
+As can be seen, two entities (`suite.12` and `other.9`) of the same type (`room`) within the same FIWARE service (`hotel`). `NGSIRestHandler` will create two `NGSILDEvent`'s:
+
+
+```
+ngsi-event-1={
+   headers={
+	   fiware-service=hotel,
+	   transaction-id=1234567890-0000-1234567890,
+	   correlation-id=1234567890-0000-1234567890,
+	   context=https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld,
+	   timestamp=1234567890,
+	},
+   original-context-element={
+	   entityId=suite.12,
+	   entityType=room,
+	   attributes=[
+	      ...
+	   ]
+	}
+}
+    
+ngsi-event-2={
+   headers={
+	   fiware-service=hotel,
+	   transaction-id=1234567890-0000-1234567890,
+	   correlation-id=1234567890-0000-1234567890,
+	   timestamp=1234567890,
+
+   },
+   original-context-element={
+	   entityId=other.9,
+	   entityType=room,
+	   attributes=[
+	      ...
+	   ]
+	}
+}
+```
+
+[Top](#top)
+
+## <a name="section2"></a>Administration guide
+### <a name="section2.1"></a>Configuration
+`NGSIRestHandler` is configured through the following parameters:
+
+| Parameter | Mandatory | Default value | Comments |
+|---|---|---|---|
+| notification\_target | no | `notify/` | Any other configured value must start with `/`. |
+| default\_service | no | `default` | Alphanumerics and underscores are only accepted. | 
+| ngsi\_version | yes | `v2` | ld for NGSI-LD and v2 for NGSIv2 notifications are only accepted. | 
+
+A configuration example could be:
+
+    cygnus-ngsi.sources = http-source
+    ...
+    cygnus-ngsi.sources.http-source.handler = com.telefonica.iot.cygnus.handlers.NGSIRestHandler
+    cygnus-ngsi.sources.http-source.notification_target = /notify
+    cygnus-ngsi.sources.http-source.default_service = default
+    cygnus-ngsi.sources.http-source.handler.ngsi_version = ld
+
+
+[Top](#top)
+
+### <a name="section2.2"></a>Accepted Content-Type
+This handler for NGSI-LD works with a `Content-Type` header with `application/json; charset=utf-8`, `application/json` and `application/ld+json`as value. Any other content type wont be considered and the notification will be discarded.
+
+It is expected UTF-8 character set is maintained by all the Flume elements in the configuration, in order the final sinks (or their backend abstractions, if they exist) compose their writes/inserts/upserts by properly specifying this kind of encoding.
+
+[Top](#top)
+
+## <a name="section3"></a>Programmers guide
+### <a name="section3.1"></a>`NGSIRestHandler` class
+TBD
```

Comment:
Pending to be completed?

Reply:
Yes, we have not prepared this section yet

Reply:
Similarly, I have checked that https://github.com/telefonicaid/fiware-cygnus/blob/master/doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_rest_handler.md#section3.1 doesn't include this section.

```suggestion
TBD (as in doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_rest_handler.md)
```